src/main/java/org/opensearch/timeseries/util/CrossClusterConfigUtils.java

```diff
@@ -0,0 +1,102 @@
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.cluster.service.ClusterService;
+
+public class CrossClusterConfigUtils {
+    private static final Logger logger = LogManager.getLogger(ParseUtils.class);
+
+    /**
+     * Uses the clusterName to determine whether the target client is the local or a remote client,
+     * and returns the appropriate client.
+     * @param clusterName The name of the cluster to evaluate.
+     * @param client The local {@link NodeClient}.
+     * @param localClusterName The name of the local cluster.
+     * @return The local {@link NodeClient} for the local cluster, or a remote client for a remote cluster.
+     */
+    public static Client getClientForCluster(String clusterName, Client client, String localClusterName) {
+        return clusterName.equals(localClusterName) ? client : client.getRemoteClusterClient(clusterName);
+    }
+
+    /**
+     * Uses the clusterName to determine whether the target client is the local or a remote client,
+     * and returns the appropriate client.
+     * @param clusterName The name of the cluster to evaluate.
+     * @param client The local {@link NodeClient}.
+     * @param clusterService Used to retrieve the name of the local cluster.
+     * @return The local {@link NodeClient} for the local cluster, or a remote client for a remote cluster.
+     */
+    public static Client getClientForCluster(String clusterName, Client client, ClusterService clusterService) {
+        logger.info("clusterName1: " + clusterName);
+        logger.info("clusterService.getClusterName().value(): " + clusterService.getClusterName().value());
+
+        return getClientForCluster(clusterName, client, clusterService.getClusterName().value());
+    }
+
+    /**
+     * Parses the list of indexes into a map of cluster_name to List of index names
+     * @param indexes A list of index names in cluster_name:index_name format.
+     *      Local indexes can also be in index_name format.
+     * @param clusterService Used to retrieve the name of the local cluster.
+     * @return A map of cluster_name:index names
+     */
+    public static HashMap<String, List<String>> separateClusterIndexes(List<String> indexes, ClusterService clusterService) {
+        return separateClusterIndexes(indexes, clusterService.getClusterName().value());
+    }
+
+    /**
+     * Parses the list of indexes into a map of cluster_name to list of index_name
+     * @param indexes A list of index names in cluster_name:index_name format.
+     * @param localClusterName The name of the local cluster.
+     * @return A map of cluster_name to List index_name
+     */
+    public static HashMap<String, List<String>> separateClusterIndexes(List<String> indexes, String localClusterName) {
+        HashMap<String, List<String>> output = new HashMap<>();
+        for (String index : indexes) {
+            String clusterName = parseClusterName(index);
+            String indexName = parseIndexName(index);
```

Comment:
Can you make one call and return a Pair instead of calling similar logic twice?

Reply:
Changed